phoenix-core-client/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -1945,6 +1948,150 @@ public KeySlots visitLeave(InListExpression node, List<KeySlots> childParts) {
       return newKeyParts(childSlot, node, new ArrayList<KeyRange>(ranges));
     }
 
+    /**
+     * If {@link ArrayAnyComparisonExpression} is of the form:
+     *
+     * <pre>
+     * COL = ANY(ARR)
+     * </pre>
+     *
+     * then we can extract the scan ranges for the COL, given COL is a PK column. This syntactical
+     * pattern can be used as a replacement for a IN expression. So, instead of following IN
+     * expression:
+     *
+     * <pre>
+     * COL IN (VAL1, VAL2, ... VALN)
```

Comment:
I don't think we can use this for RVC list like (COL1, COL2) IN ( (val11,val12), (val21, val22))

Reply:
Yes, using Array Any we can't use RVC list as we don't support 2D arrays. I tried this.